src/resolvers/Mutation/createPost.ts

```diff
@@ -93,6 +95,47 @@ export const createPost: MutationResolvers["createPost"] = async (
     imageUrl: args.file ? uploadImageObj?.newImagePath : "",
   });
 
+  // Check if the pinned property is set
+  if (args.data.pinned) {
+    // Check if the user has privileges to pin the post
+    const currentUserIsOrganizationAdmin = currentUser.adminFor.some(
+      (organizationId) => organizationId.toString() === args.data.organizationId
+    );
+
+    if (
+      !(currentUser!.userType === "SUPERADMIN") &&
+      !currentUserIsOrganizationAdmin
+    ) {
+      throw new errors.UnauthorizedError(
+        requestContext.translate(USER_NOT_AUTHORIZED_TO_PIN.message),
+        USER_NOT_AUTHORIZED_TO_PIN.code,
+        USER_NOT_AUTHORIZED_TO_PIN.param
+      );
+    }
+
+    // Add the post to pinnedPosts of the organization
+    await Organization.updateOne(
+      { _id: args.data.organizationId },
+      {
+        $push: {
+          pinnedPosts: createdPost._id,
+        },
+      }
+    );
+
+    // Change the pinned status of the post
+    await Post.updateOne(
```

Comment:
Arrange the checks in a way that `pinned` is set in the same step the post is created.